src/test/java/seedu/ddd/model/person/IdTest.java

```diff
@@ -0,0 +1,65 @@
+package seedu.ddd.model.person;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.ddd.testutil.Assert.assertThrows;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.ddd.model.contact.common.Id;
+
+public class IdTest {
+
+    @Test
+    public void constructor_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new Id(null));
+    }
+
+    @Test
+    public void constructor_invalidAddress_throwsIllegalArgumentException() {
+        String invalidAddress = "";
+        assertThrows(IllegalArgumentException.class, () -> new Id(invalidAddress));
+    }
+
+    @Test
+    public void isValidAddress() {
+        // null address
+        assertThrows(NullPointerException.class, () -> Id.isValidId(null));
+
+        // invalid ids
+        assertFalse(Id.isValidId(-1));
+        assertFalse(Id.isValidId(-10));
+        assertFalse(Id.isValidId(""));
+        assertFalse(Id.isValidId(" "));
+
+        // valid ids
+        assertTrue(Id.isValidId(0));
+        assertTrue(Id.isValidId(1));
+        assertTrue(Id.isValidId(10));
+        assertTrue(Id.isValidId(100));
+        assertTrue(Id.isValidId("0"));
+        assertTrue(Id.isValidId("1"));
+        assertTrue(Id.isValidId("10"));
+        assertTrue(Id.isValidId("100"));
```

Comment:
You mean add e.g. `INVALID_ID_0` as constants in `CommandTestUtil`? I think it's ok to leave it here because `NameTest`, `EmailTest` etc. just use hardcoded strings, and `Id` validation logic isn't part of command logic.